src/seedu/addressbook/Main.java

```diff
@@ -105,12 +118,21 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
+    /**
+     * Executes the command and returns the result.
+     *
+     * @param command user command
+     * @return result of the command
+     */
+    private CommandResult executeCommand(Command command) {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException e) {
+            ui.showToUser(e.getMessage());
+            throw new RuntimeException(e);
```

Comment:
Just handle the exception here. There is no need to throw it back to its caller method. Otherwise you just throw this error to another function, which will still cause your program to crash. (For this LO, after you catch the exception, it is fine to just print something on the screen.)